cmd/queue/main.go

```diff
@@ -319,8 +321,38 @@ func main() {
 			}
 		})
 
+		flush(logger)
 		if err := adminServer.Shutdown(context.Background()); err != nil {
 			logger.Errorw("Failed to shutdown admin-server", zap.Error(err))
 		}
 	}
 }
+
+func pushRequestLogHandler(currentHandler http.Handler) http.Handler {
+	templ := os.Getenv("SERVING_REQUEST_LOG_TEMPLATE")
+	if templ == "" {
+		return currentHandler
+	}
+
+	revInfo := &queue.RequestLogRevInfo{
+		Name:          os.Getenv("SERVING_REVISION"),
+		Namespace:     os.Getenv("SERVING_NAMESPACE"),
+		Service:       os.Getenv("SERVING_SERVICE"),
+		Configuration: os.Getenv("SERVING_CONFIGURATION"),
+		PodName:       os.Getenv("SERVING_POD"),
+		PodIP:         os.Getenv("SERVING_POD_IP"),
+	}
+	handler, err := queue.NewRequestLogHandler(currentHandler, utils.NewSyncFileWriter(os.Stdout), templ, revInfo)
+
+	if err != nil {
+		logger.Errorw("Error setting up request logger. Request logs will be unavailable.", zap.Error(err))
```

Comment:
It looks like this can really only fail if I have a bad template.

I wonder if this should cause the container to fail startup rather than continue without request logging. Looking at the rest of the initialization logic for the queue proxy if we cannot get settings from users we choose to break rather than fall back.